source/index.js

```diff
@@ -146,7 +173,14 @@ module.exports = async (input = 'patch', options) => {
 						return `Private package: not publishing to ${pkgManagerName}.`;
 					}
 				},
-				task: (context, task) => publishTaskHelper(pkgManager, task, options, input)
+				task: async (context, task) => {
+					try {
+						await publishTaskHelper(pkgManager, task, options, input);
+						published = true;
+					} catch (_) {
+						rollback();
```

Comment:
For some reason, sometimes `rollback` gets called and sometimes it doesn't (when `publishTaskHelper` throws an error).

Reply:
I did fix some problems in https://github.com/sindresorhus/np/commit/304030e29d8744ab9a11fc1b76d9923b68a86c56